nucliadb_models/src/nucliadb_models/search.py

```diff
@@ -389,6 +390,40 @@ class SortOptions(BaseModel):
     order: SortOrder = SortOrder.DESC
 
 
+class RankFusionName(str, Enum):
+    LEGACY = "legacy"
+    RECIPROCAL_RANK_FUSION = "rrf"
+
+
+class _BaseRankFusion(BaseModel):
+    name: str
+
+
+class LegacyRankFusion(_BaseRankFusion):
+    name: Literal[RankFusionName.LEGACY] = RankFusionName.LEGACY
+
+
+class ReciprocalRankFusion(_BaseRankFusion):
+    name: Literal[RankFusionName.RECIPROCAL_RANK_FUSION] = RankFusionName.RECIPROCAL_RANK_FUSION
+    k: float = Field(
+        default=60.0,
+        title="RRF k parameter",
+        description="k parameter changes the influence top-ranked and lower-ranked elements have. Research has shown that 60 is a performant value across datasets",  # noqa: E501
+    )
+    boosting: dict[Union[Literal["keyword"], Literal["semantic"]], float] = Field(
+        default_factory=partial(dict, keyword=1.0, semantic=1.0),
+        title="Retriever boosting",
+        description="Define weights for each retriever. This allows to give different priorities to different retrieval methods. Scores will be multiplied by this value, so a boost of 1.0 is the same as no boost. As an example, multilingual search may benefit from semantic boosting",  # noqa: E501
+        min_length=2,
+    )
```

Comment:
Is this too hacky?

Reply:
why not definig a new model with validation? 
I would put defaults to 1 to both keyword and semantic, and then compute the relative weights between both.

This way if the user specifies only one of them, the other is taken as 1.